esphome/components/wifi/__init__.py

```diff
@@ -272,6 +274,8 @@ def _validate(config):
             cv.SplitDefault(CONF_OUTPUT_POWER, esp8266=20.0): cv.All(
                 cv.decibel, cv.float_range(min=8.5, max=20.5)
             ),
+            cv.Optional(CONF_BTM, default=False): cv.boolean,
```

Comment:
```suggestion
            cv.Optional(CONF_BTM, default=False): cv.All(cv.only_on_esp32, cv.only_with_esp_idf, cv.boolean),
```

To only allow this on ESP32 with esp-idf

And you can use `cv.SplitDefault` to only set the default when on this platform.

Reply:
Hmmm I must have done something wrong, as there's a failing test now.

Reply:
Nvm, think I found it